src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonWebToken.PayloadClaimSet.cs

```diff
@@ -73,8 +74,15 @@ internal JsonClaimSet CreatePayloadClaimSet(byte[] bytes, int length)
                     }
                     else if (reader.ValueTextEquals(JwtPayloadUtf8Bytes.Iss))
                     {
-                        _iss = JsonSerializerPrimitives.ReadString(ref reader, JwtRegisteredClaimNames.Iss, ClassName, true);
-                        claims[JwtRegisteredClaimNames.Iss] = _iss;
+                        IssuerUtf8 = Encoding.UTF8.GetBytes(JsonSerializerPrimitives.ReadString(ref reader, JwtRegisteredClaimNames.Iss, ClassName, true));
```

Comment:
Better way to get bytes from stream?

Reply:
this is going to be wasted work, utf8JsonReader -> utf16 string -> utf8 bytes.

It is a larger refactor, but would be curious what the perf would look like if we used: https://learn.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.trygetbytesfrombase64?view=net-8.0

Reply:
Do you see a better way to directly get bytes from a utf8JsonReader?

Reply:
Yea, this is just a placeholder code. Was looking for a good way to get bytes out of the reader.
Seems like ValueSpan has some [restrictions](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.valuespan?view=net-8.0#remarks):

> There is no guarantee that this property will always contain well-formed data. If the input JSON passed in to the Utf8JsonReader contains invalid UTF-8 bytes within JSON string tokens, this property will return back those invalid UTF-8 bytes as is. Therefore, if the input is untrusted or not previously validated, call [GetString()](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.getstring?view=net-8.0#system-text-json-utf8jsonreader-getstring) to get the JSON string token.
> 
> Since this property returns the raw bytes, avoid using it for text comparison. Instead call [ValueTextEquals](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.valuetextequals?view=net-8.0) which unescapes the text if necessary.

Reply:
So if the validation is done [here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/051d164e3c025a0d7276f1d6acf38c902a4893fc/src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs#L848), I assume we'd have to refactor that to be a be able to use that method here (since it requires a token and a TVPs).